internal/ykclient/api_handlers.go

```diff
@@ -139,18 +163,20 @@ func (c *Client) GetApplications(ctx context.Context, partitionName, queueName s
 			return nil, fmt.Errorf("could not unmarshal application from response: %v", err)
 		}
 
-		for _, a := range responseApps {
-			apps = append(apps, &a)
+		for i := range responseApps {
+			apps = append(apps, &responseApps[i])
 		}
 	}
 
 	return apps, nil
 }
 
-// GetApplication calls the YuniKorn Scheduler API to get the application information for the given partition, queue and appID
-// If partitionName is empty, it defaults to "default"
+// GetApplication calls the YuniKorn Scheduler API to get the application information for the given
+// partition, queue and appID. If partitionName is empty, it defaults to "default".
```

Comment:
Could we express this comment in the form of a unit test?

Reply:
I will create a test.

Reply:
Test added in commit c3a3e73 - see `internal/ykclient/api_handlers_test.go`.